source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ContainerRenderer.java

```diff
@@ -226,6 +229,17 @@ public static void setSelectAction(RenderedAdaptiveCard renderedCard, BaseAction
             }
 
             view.setOnClickListener(new BaseActionElementRenderer.SelectActionOnClickListener(renderedCard, selectAction, cardActionHandler));
+
+            String contentDescription = !TextUtils.isEmpty(selectAction.GetTitle()) ? selectAction.GetTitle() : selectAction.GetTooltip();
+            String tooltip = !TextUtils.isEmpty(selectAction.GetTooltip()) ? selectAction.GetTooltip() : selectAction.GetTitle();
+            if (!TextUtils.isEmpty(tooltip))
+            {
+                view.setContentDescription(contentDescription);
+            }
+            if (!TextUtils.isEmpty(tooltip))
+            {
+                TooltipCompat.setTooltipText(view, tooltip);
+            }
```

Comment:
feels like this should probably be pulled out into a helper function

Reply:
Refactored